test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs

```diff
@@ -57,5 +72,53 @@ public void DthStartup_GetProjectInformation(string flavor, string os, string ar
                 Assert.Contains("net46", frameworkShorNames);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DthStartup_ProtocolNegotiation(string flavor, string os, string architecture)
+        {
+            var projectName = "EmptyConsoleApp";
+            var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+            var testProject = _fixture.GetTestProjectPath(projectName);
+
+            using (var server = DthTestServer.Create(runtimeHomePath, testProject))
+            using (var client = new DthTestClient(server, 0))
+            {
+                for (int testVersion = 1; testVersion <= 3; ++testVersion)
```

Comment:
I would split this up into different tests so the assertions are more clear. This could be a helper that takes the protocolVersion and the expectedProtocol version returned.

```
DthStartup_ProtocolNegotiation(flavor, os, architecture, protocolVersion: 1, expectedProtocolVersion: 2);
```